tensorflow/core/ops/math_ops.cc

```diff
@@ -131,90 +131,102 @@ tf.complex_abs(x) ==> [5.25594902, 6.60492229]
 
 // Declares cwise unary operations signature: 't -> 't
 #define UNARY()                      \
-  Input("x: T").Output("y: T").Attr( \
-      "T: {half, float, double, int32, complex64, int64}")
+  Input("x: T").Output("y: T")
```

Comment:
Thanks! Then, let me submit two separate PRs:

1. adding the `complex128` support
2. adding 'UNARY' MACROS

for easy review and testing. I'm working on it.